cmd/convox/env.go

```diff
@@ -166,27 +175,46 @@ func cmdEnvSet(c *cli.Context) error {
 		data += fmt.Sprintf("%s\n", value)
 	}
 
-	fmt.Print("Updating environment... ")
+	output := os.Stdout
+	if c.Bool("id") {
+		output = os.Stderr
+	}
+
+	output.Write([]byte("Updating environment... "))
 
 	_, releaseID, err := rackClient(c).SetEnvironment(app, strings.NewReader(data))
 	if err != nil {
 		return stdcli.Error(err)
 	}
 
-	fmt.Println("OK")
+	output.Write([]byte("OK\n"))
 
 	if releaseID != "" {
 		if c.Bool("promote") {
-			fmt.Printf("Promoting %s... ", releaseID)
+			output.Write([]byte(fmt.Sprintf("Promoting ")))
+			os.Stdout.Write([]byte(fmt.Sprintf("%s", releaseID)))
 
 			_, err = rackClient(c).PromoteRelease(app, releaseID)
 			if err != nil {
 				return stdcli.Error(err)
 			}
 
-			fmt.Println("OK")
+			// If we're redirecting stdout but not stderr, print the release ID to stderr
+			if c.Bool("id") && terminal.IsTerminal(int(os.Stderr.Fd())) && !terminal.IsTerminal(int(os.Stdout.Fd())) {
```

Comment:
As I recall from my manual testing, something like this is needed because otherwise if the user redirects the output to a file, stderr will print something like

```
To deploy these changes run `convox releases promote `
```
 without the release ID (and/or the release ID will get written to stdout twice.)